docs/dev/utils.rst

```diff
@@ -0,0 +1,24 @@
+Other Utilities and Customizations
+==================================
+
+The ESP website has a bunch of other useful things in it that are useful to know.  Here are some of them:
+
+Rendering and Templates
+-----------------------
+* We use a modified ``esp.web.util.main.render_to_response`` instead of django's; it takes the template path, the request, and the context, and inserts useful things into the context, like information needed to render navbars, and the request.
+* To render a block of text that will be editable by admins, put a ``{% load render_qsd %}`` somewhere, and then put a ``{% render_inline_qsd name %}`` or ``{% render_inline_program_qsd program name %}`` in where you want the block.  If you want it to have default text, put it between the tags ``{% inline_qsd_block name %}`` and ``{% end_inline_qsd_block %}`` or ``{% inline_program_qsd_block program name %}`` and ``{% end_inline_program_qsd_block %}``.
+* You almost always want your template to begin with ``{% extends "main.html" %}``, to inherit from the main site layout; then put the content between ``{% block content %}`` and ``{% endblock %}`` tags.
```

Comment:
In the cases where it doesn't make sense to extend "main.html" or one of its extentions, then you almost certainly want to extend "elements/html".

Can also use ``{% endblock content %}`` for added clarity.